docs/pip_transition.md

```diff
@@ -0,0 +1,79 @@
+  
+  
+Steps
+
+### 1.  Build python packages and add them to dr
+* Building:
+   * py2deb seems to work reasonably well. The only caveat is that it needs `pip-accel` to be manually installed first `pip install pip-accel` as otherwise you get a version of pip which pip-accel is not compatible with. Then do `pip install py2deb`.
+* Choice of packages and versions:
+  * It is not necessary to install pip once we transition.
+  * Some packages also already have raspbian versions. I propose to build all of them to the exact versions we used in 3.14.1, and for the next release pin those version of the deb packages. This is to reduce the QA burden of the next release. For future releases we can unpin them and unpush deb packages which have versions in Raspbian.
+  There are only two where the raspbian version is newer (pam and futures).
+  
+  
+  
+  * Although some people may have updated and got requests 2.18.4, the last release had 2.6.0 so I still propose to use that.
+    The reason is that 2.18.4 installs a bunch of other stuff.
+
+
+
+
+### 2. Transition from pip to deb packages.
+
+This must:
+
+  * Install the new deb packages
+  * Ensure the old pip packages don't override them.
+  
+It turns out by default pip packages do override deb packages. Python load modules from `sys.path`
+
+* deb installs to `/usr/lib/python2.7/dist-packages`
+* pip installs to `/usr/local/lib/python2.7/dist-packages`
+
+`sys.path` is set up so that pip has priority. Therefore we need to make sure that the deb files
+get priority. My suggested fix is to add a new path `/usr/local/lib/python2.7/dist-packages.pip-fallback` later in the sys.path list, and then move
+the current pip files there. In the event of a failure of some kind, python will still work.
+
+
+#### Transition update process
+##### 1 `python-docopt` installed as a (new) dependency of kano-updater
+
+##### 2 Install fallback path
+ `kano-updater.install` , adds a file
+   `/usr/lib/python2.7/dist-packages/kano-pip-compat.pth`
+ containing `/usr/local/lib/python2.7/dist-packages.pip-fallback`
+This adds that path to sys.path after `/usr/lib/python2.7/dist-packages`.
+
+##### 3 Pre-update scenario Move old pip packages to fallback path 
+```
+mkdir -p /usr/local/lib/python2.7/dist-packages.pip-fallback
+mv /usr/local/lib/python2.7/dist-packages{,.pip-fallback}
+mkdir -p /usr/local/lib/python2.7/dist-packages # leave empty directory (just in case)
+```
+##### 4 deb packages installed as part of main install step
+A new package `kano-os-pip-transition-depends` which depends on the new deb packages will be added.
+
+The code in the updater which downloads/installs pip packages needs to be removed before this update.
+
+### 3. Pay down tech debt:
+ * Add correct dependencies to other packages.
+ * Stop shipping our built python deb files once newer ones become available in raspbian\
+ 
```

Comment:
I believe that it would be easier for us to always ship our own python `deb`'s as that way we can always ensure that we have tested the required versions and are not dependent on the Raspbian releases.

Reply:
That might be true. We should definitely build them for now, anyway, so that's a decision for later.